### DISCUSSION TITLE
System.Memory correct version missing

### DISCUSSION THREAD

**Discussion by betsegaw** [spectreconsole/spectre.console#670] [Q&A]
**Information**
 - OS: Windows
 - Version: 0.43.1-preview.0.16
 - Terminal: Windows Terminal

**Describe the bug**
Loading nuget package in Powershell and trying to use it results in an exception with `Could not load file or assembly 'System.Memory...`

**To Reproduce**
1. Download the nuget package into a folder
2. Load it into Powershell with command `[System.Reflection.Assembly]::LoadFrom("/path/to/lib/netstandard2.0/Spectre.Console.dll")`
3. Try to use it with `[Spectre.Console.AnsiConsole]::Markup("[underline red]Hello[/] World!")`

**Expected behavior**
Nicely formatted output

**Actual output**
```
Exception calling "Markup" with "1" argument(s): "Could not load file or assembly 'System.Memory, Version=4.0.1.0,
Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file
specified."
At line:1 char:1
+ [Spectre.Console.AnsiConsole]::Markup("[underline red]Hello[/] World! ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FileNotFoundException
```

**Additional context**
It seems that this is a lack of binding redirects that ensure that the nuget package can use the latest version rather than lock onto a specific version? 

Also, might be resolved by including the following

```xml
<AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
<GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
```

**Comment by nils-a** [comment]
@betsegaw No, *BindingRedirects* are all automatic these days. If those are needed, they need to be created at the entry-point of the application.

The code you showed above works fine in PowerShell 7.

Now, if you were on an old PowerShell 5.x - that's probably .NET Framework 4.6 or something. You'd need to add binding-redirects manually to PowerShell.

Something like 
```powershell
nuget install system.memory -outputdirectory .
$memory = [reflection.assembly]::LoadFrom((Resolve-Path .\System.Memory.4.5.4\lib\netstandard2.0\System.Memory.dll))
$redirect = [ResolveEventHandler] {
  param($sender, $e)

  if ($e.Name -like "System.Memory,*") { return $memory }

  return $null
}
[appdomain]::CurrentDomain.add_AssemblyResolve($redirect)
```

Should get you on the right track, note that this is probably not complete: `System.Memory` comes with additional dependencies, which need probably also be added to the redirect.
